src/scf-release/src/acceptance-tests-brain/test-scripts/testutils.rb

```diff
@@ -195,15 +195,18 @@ def run_with_retry(retries, interval)
 # Poll the status of a Kubernetes namespace, until all the pods in that
 # namespace are ready and all the jobs have run.
 def wait_for_namespace(namespace, sleep_duration=10)
+    STDERR.puts "#{c_bold}# Waiting for: #{namespace}, in increments of #{sleep_duration}s#{c_reset}"
     loop do
-        output = capture("kubectl get pods --namespace #{namespace} --no-headers")
-        ready = !output.empty?
-        output.each_line do |line|
-            name, readiness, status, restarts, age = line.split
-            next if status == 'Completed'
-            next if status == 'Running' && /^(\d+)\/\1$/ =~ readiness
+        output, status = capture_with_status("kubectl get pods --namespace #{namespace} --no-headers")
+        ready = status.success? && !output.empty?
+        if ready
```

Comment:
Hmm, I read this line three times and wrote two (incorrect) comments before figuring out this is actually correct.

Probably just means I'm dumb, though.

Do we need to wait if `!status.success?`? Would we not expect the kube cluster to recover relatively quickly? Or are we worried about generated load?

Reply:
Yes. Note that the original code fully aborted for `!status.success?`. Even if it was just a glitch in talking to the kube controller (See referenced CAP-1043). The change is to recover from the glitch by simply waiting as if the namespace is not ready.